packages/db/tests/models/test-granule-model.js

```diff
@@ -252,3 +253,89 @@ test('GranulePgModel.upsert() will allow a running status to replace a completed
   const record = await granulePgModel.get(knex, { granule_id: granule.granule_id });
   t.is(record.status, 'running');
 });
+
+test('GranulePgModel.delete() deletes an unpublished granule', async (t) => {
+  const {
+    knex,
+    granulePgModel,
+    collectionCumulusId,
+    executionCumulusId,
+  } = t.context;
+
+  const granule = fakeGranuleRecordFactory({
+    status: 'completed',
+    collection_cumulus_id: collectionCumulusId,
+    execution_cumulus_id: executionCumulusId,
+    published: false,
+  });
+
+  await granulePgModel.create(knex, granule);
+
+  t.like(
+    await granulePgModel.get(knex, granule),
+    granule
+  );
+
+  await granulePgModel.delete(knex, granule);
+
+  await t.throwsAsync(
+    granulePgModel.get(knex, { granule_id: granule.granule_id }),
+    { instanceOf: RecordDoesNotExist }
+  );
+});
+
+test('GranulePgModel.delete() throws an error if the granule is published', async (t) => {
+  const {
+    knex,
+    granulePgModel,
+    collectionCumulusId,
+    executionCumulusId,
+  } = t.context;
+
+  const granule = fakeGranuleRecordFactory({
+    status: 'completed',
+    collection_cumulus_id: collectionCumulusId,
+    execution_cumulus_id: executionCumulusId,
+    published: true,
+  });
+
+  await granulePgModel.create(knex, granule);
+
+  try {
+    granulePgModel.delete(knex, granule);
+  } catch (error) {
+    t.is(error.message, 'You cannot delete a granule that is published to CMR. Remove it from CMR first');
+  }
```

Comment:
I checked out the code and `t.throws` would work, however it has a different syntax:

`t.throws(() => granulePgModel.delete(knex, granule));`

However, I think it would be preferable to mark the delete method on the granule model and base model as `async` and then handle with `throwsAsync`. Unless we mark these methods as async, I don't think we're going to get the benefit of promise based error handling

Reply:
Yup, that makes sense. Updated the delete method on BasePgModel and GranulePgModel here:

https://github.com/nasa/cumulus/pull/2057/commits/ca589938b99585342cf0c29045edd6587f5888bd